src/inventory.cpp

```diff
@@ -425,7 +425,7 @@ void InventoryList::setName(const std::string &name)
 	setModified();
 }
 
-void InventoryList::serialize(std::ostream &os, bool incremental) const
+void InventoryList::serialize(std::ostream &os, bool incremental, bool disk) const
```

Comment:
`disk` partially conflicts with the intentions of `incremental`.
Either
1. rename & clarify `disk` to `sparse_metadata` (or similar)
2. or substitute `incremental` within this function with `!disk` because it is an optimization that's always performed over network, but never to disk.

Same concerns for Inventory::serialize

Reply:
2. is not a good option since incremental can be false even when disk is false - in fact, incremental seems to never be true since incremental inventory sending is unimplemented server-side.